jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -580,12 +588,15 @@ private void checkSpecialAccess(Class<?> callerClass) throws IllegalAccessExcept
 		 */
 		public MethodHandle findSpecial(Class<?> clazz, String methodName, MethodType type, Class<?> specialToken) throws IllegalAccessException, NoSuchMethodException, SecurityException, NullPointerException {
 			nullCheck(clazz, methodName, type, specialToken);
-			checkSpecialAccess(specialToken);	/* Must happen before method resolution */
+			checkSpecialAccess(clazz, specialToken);	/* Must happen before method resolution */
 			MethodHandle handle = null;
 			try {
 				handle = findSpecialImpl(clazz, methodName, type, specialToken);
 				Class<?> handleDefc = handle.getDefc();
-				if ((handleDefc != accessClass) && !handleDefc.isAssignableFrom(accessClass)) {
+				/* Check the relationship between the lookup class and the current class
+				 * only when the requested caller class is the same as the lookup class.
+				 */
+				if ((specialToken == accessClass) && !handleDefc.isAssignableFrom(accessClass)) {
```

Comment:
I'm not sure I follow this logic.  The specialToken could be a subclass of the accessClass.  I would still expect it to have to be assignable in that case.

Reply:
I need to double-check the case to see whether it works for all existing related test cases.

Reply:
I already verified all JSR292 related test case in personal build on Java 8 & 11 and nothing wrong was detected. So updated to `accessClass.isAssignableFrom(specialToken)` as specialToken should be assignable to the accessClass in our case.